modules/tensor_mechanics/src/auxkernels/RadialDisplacementAux.C

```diff
@@ -0,0 +1,212 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "RadialDisplacementAux.h"
+#include "MooseMesh.h"
+
+template<>
+InputParameters validParams<RadialDisplacementAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+  params.addClassDescription("Compute the radial component of the displacement vector for cylindrical or spherical models.");
+  params.addRequiredCoupledVar("displacements", "The displacements appropriate for the simulation geometry and coordinate system");
+  params.addParam<RealVectorValue>("axis_point_1", "Start point for line defining cylindrical axis (3D Cartesian models)");
+  params.addParam<RealVectorValue>("axis_point_2", "End point for line defining cylindrical axis (3D Cartesian models)");
+  params.addParam<RealVectorValue>("origin", "Sphere origin for 3D Cartesian and 2D axisymmetric models, or circle origin for 2D Cartesian models");
+  params.set<bool>("use_displaced_mesh") = false;
+
+  return params;
+}
+
+RadialDisplacementAux::RadialDisplacementAux(const InputParameters & parameters) :
+  AuxKernel(parameters),
+  _have_axis(false),
+  _have_origin(false)
+{
+  const std::set<SubdomainID> & subdomains = _mesh.meshSubdomains();
+  const auto & sbd_begin = *subdomains.begin();
+  for (const auto & sbd : subdomains)
+  {
+    if (sbd == sbd_begin)
+      _coord_system = _subproblem.getCoordSystem(sbd);
+    else if (_subproblem.getCoordSystem(sbd) != _coord_system)
+      mooseError("RadialDisplacementAux requires that all subdomains have the same coordinate type");
+  }
```

Comment:
This loop should be factored out. I just wrote one like it in the tensor mechanics action.